tests/test_trade.py

```diff
@@ -22,139 +22,136 @@ class SingleTradeTests(unittest.TestCase):
-                    market_state=simulator.market.market_state, share_price=simulator.market.market_state.share_price
-                )
-                assert np.allclose(total_liquidity_direct, total_liquidity_agent, atol=0, rtol=1e-15), (
-                    f"test_trade.test_compare_agent_to_calc_liquidity: ERROR: {total_liquidity_direct=}"
-                    f"does not equal {total_liquidity_agent=} "
-                    f"off by {(np.abs(total_liquidity_direct - total_liquidity_agent))=}."
-                )
-                assert np.allclose(market_direct.apr, simulator.market.apr, atol=0, rtol=1e-12), (
-                    f"test_trade.test_compare_agent_to_calc_liquidity: ERROR: {market_direct.apr=}"
-                    f" does not equal {simulator.market.apr=}"
-                    f"off by {(np.abs(market_direct.apr - simulator.market.apr))=}."
-                )
+                for num_position_days in [90, 365]:
+                    for pricing_model_name in ["Yieldspace", "Hyperdrive"]:
+                        # run_base_trade_test initializes a market with an LP agent
+                        config = Config()
+                        config.pricing_model_name = pricing_model_name
+                        config.target_liquidity = target_liquidity
+                        config.trade_fee_percent = 0.1
+                        config.redemption_fee_percent = 0.0
+                        config.target_pool_apr = target_pool_apr
+                        config.num_trading_days = 3  # sim 3 days to keep it fast for testing
+                        config.num_blocks_per_day = 3  # 3 block a day, keep it fast for testing
+                        config.vault_apr = [0.05] * config.num_trading_days
+                        config.num_position_days = num_position_days  # how long until token maturity
+                        simulator = sim_utils.get_simulator(config)
+                        logging.debug(
+                            (
+                                "\n\n----\n"
+                                "target_liquidity=%g\n"
+                                "target_pool_apr=%g\n"
+                                "num_position_days=%g\n"
+                                "pricing_model_name=%s\n"
+                                "simulator.market.market_state=%s"
+                            ),
+                            target_liquidity,
+                            target_pool_apr,
+                            num_position_days,
+                            pricing_model_name,
+                            simulator.market.market_state,
+                        )
+                        # assign the results of the init_lp agent to explicit variables
+                        # market_state is used only for share_price and init_share_price
+                        share_reserves_direct, bond_reserves_direct = simulator.market.pricing_model.calc_liquidity(
```

Comment:
this is the exact method that is used in market_state, so we aren't really verifying values.  In cases like this it's better to handroll and do a hardcoded value in case the underlying function is changed and messed up at some point.

Reply:
or rather, this and init market both call calc_bond_reserves.